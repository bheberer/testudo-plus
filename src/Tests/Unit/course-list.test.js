import React from 'react';
import { shallow, mount } from 'enzyme';
import CourseList from '../../Components/course-list';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
// import IconButton from '@material-ui/core/IconButton';
// import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
// import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import Paper from '@material-ui/core/Paper';
// import List from '@material-ui/core/List';
// import ListSubHeader from '@material-ui/core/ListSubheader';
// import ListItem from '@material-ui/core/ListItem';
// import ListItemText from '@material-ui/core/ListItemText';

jest.mock('@material-ui/core/Paper', () => 'mock-paper')

describe('course list integration tests', () => {

  it('renders correctly', () => {
    const tree = mount(
      <CourseList courses={ [
        { course_id: 'CMSC131', name: 'CS 101' }, 
        { course_id: 'MATH140', name: 'Calc 1'}, 
        { course_id: 'UNIV100', name: 'university 100'}, 
        { course_id: 'ASTR100', name: 'astronomy 1' }
      ] } />
    )

    expect(toJson(tree, { mode: 'shallow' })).toMatchSnapshot();
    tree.unmount();
  })

  it('renders correct courses', () => {
    const tree = mount(
      <CourseList courses={ [
        { course_id: 'CMSC131', name: 'CS 101' }, 
        { course_id: 'MATH140', name: 'Calc 1'}, 
        { course_id: 'UNIV100', name: 'university 100'}, 
        { course_id: 'ASTR100', name: 'astronomy 1' }
      ] } />
    )

    // const treeJson = toJson(tree);

    // const courses = tree.find('ListItemText');

    expect(tree).toBe({});



    //tree = CourseList({ courses: ['CMSC131', 'MATH140', 'UNIV100', 'ASTR100'] })
    // expect(courses).toEqual(['CMSC131', 'MATH140', 'UNIV100', 'ASTR100']);
  })
})
