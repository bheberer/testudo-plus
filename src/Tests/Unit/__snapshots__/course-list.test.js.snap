// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`course list integration tests renders correctly 1`] = `
<CourseList
  classes={
    Object {
      "root": "CourseList-root-1",
      "subHeader": "CourseList-subHeader-2",
    }
  }
  courses={
    Array [
      "CMSC131",
      "MATH140",
      "UNIV100",
      "ASTR100",
    ]
  }
/>
`;
